Projects/Modules/Documentarian.MicrosoftDocs/Source/Public/Functions/Get-CmdletXref.ps1

```diff
@@ -1,52 +1,72 @@
-﻿# Copyright (c) Microsoft Corporation.
+# Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
+using namespace System.Management.Automation
+
 function Get-CmdletXref {
     <#
     .SYNOPSIS
     Gets a cross-reference link for a command.
     #>
     [CmdletBinding()]
-    [OutputType('System.String[]')]
+    [OutputType([string[]])]
     param(
-        # The name of the command to get a cross-reference link for.
+        # The name of the Command or a CommandInfo instance to get a cross-reference link for.
         [Parameter(Mandatory, ValueFromPipeline)]
-        [string[]]$Name
+        [object[]] $Command
     )
-
+    
     begin {
-        $ProgressPreference = 'SilentlyContinue'
+        $allowedFlags = [CommandTypes] 'Cmdlet, Function, Filter'
     }
-
     process {
-        foreach ($cmdname in $Name) {
+        foreach ($cmd in $Command) {
             try {
-                $cmd = Get-Command $cmdname -ErrorAction Stop
-                if ($cmd.CommandType -eq 'Alias') {
-                    Write-Verbose "$cmdname is an alias for $($cmd.ResolvedCommand)"
-                    $cmd = Get-Command $cmd.ResolvedCommand -ErrorAction Stop
+                if ($cmd -isnot [CommandInfo]) {
+                    $tryGetCommand = $PSCmdlet.InvokeCommand.GetCommand(
```

Comment:
I'm guessing the change from calling the cmdlets normally to using `$PSCmdlet.InvokeCommand` is for performance, but I don't think that's a concern for this function and the scale it is likely to be used at. Instead, would prefer using the cmdlets more idiomatically. This syntax _looks_ more complex to less experienced contributors and can affect our ability to write unit tests in Pester (not that we have much in the way of test coverage anywhere right now).

Reply:
@michaeltlombardi I think I want to keep this change. It makes the error handling easier. I don't think it will have an impact on testability.

Reply:
Unless things have changed in Pester, I don't think you can mock method invocations - we can just do acceptance testing and actually invoke the methods. I'm okay with taking it as-is, but I think the points around non-expert readability / project-idiomaticity apply. Happy to mark this one resolved.